app/experimenter/nimbus-ui/src/hooks/useReviewCheck.tsx

```diff
@@ -0,0 +1,69 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from "react";
+import { getExperiment_experimentBySlug } from "../types/getExperiment";
+
+export type ReviewCheck = ReturnType<typeof useReviewCheck>;
+
+const fieldPageMap: { [page: string]: string[] } = {
+  overview: [
+    "public_description",
+    "risk_brand",
+    "risk_revenue",
+    "risk_partner_related",
+  ],
+  branches: ["reference_branch", "treatment_branches", "feature_config"],
+  audience: [
+    "channel",
+    "firefox_min_version",
+    "targeting_config_slug",
+    "proposed_enrollment",
+    "proposed_duration",
+    "population_percent",
+    "total_enrolled_clients",
+  ],
+};
+
+type FieldReviewProps = {
+  field: string;
+  children: (props: {
+    field: string;
+    message: string;
+  }) => React.ReactNode | null;
+};
+
+export function useReviewCheck(
+  experiment: getExperiment_experimentBySlug | null | undefined,
+) {
+  const reviewItems = (experiment?.readyForReview?.message || {}) as Record<
+    string,
+    string[]
+  >;
+  const invalidFields = Object.keys(reviewItems);
+  const invalidPages = Object.keys(fieldPageMap).filter((page) =>
+    fieldPageMap[page].some((field) => invalidFields.includes(field)),
+  );
+  const fieldReviewMessages = (fieldName: string): string[] =>
+    reviewItems[fieldName] || [];
+
+  // EXP-981 should remove this component entirely once fieldReviewMessages is used
+  // to feed these review-readiness messages directly into useCommonForm
+  const FieldReview: React.FC<FieldReviewProps> = ({ field, children }) => {
+    const messages = fieldReviewMessages(field);
+    if (!messages.length || !window.location.search.includes("show-errors")) {
+      return null;
+    }
+
+    return <>{children({ field, message: messages.join(", ") })}</>;
```

Comment:
It looks like we only use this component like...

```
<FieldReview field="channel">
  {({ field, message }) => (
    <InlineErrorIcon {...{ field, message }} />
  )}
</FieldReview>
```

or like this...

```
<FieldReview field="risk_brand">
  {({ field, message }) => (
    <span className="align-text-bottom ml-1">
      <InlineErrorIcon {...{ field, message }} />
    </span>
  )}
</FieldReview>
```

I know your comment says EXP-981 should remove these in favor of getting these from `useCommonForm`, but do we really need `field, message` returned from `FieldReview`? Can we take a prop instead of accepting `children`?

```
<FieldReview field="channel" />
```
and (or whatever prop makes sense to pass in):
```
<FieldReview field="field_risk" alignBottom />
```

Reply:
@LZoog I totally agree except for the fact that in some cases we need to wrap the error icon in a span for styling. Otherwise I'd say let's make it all internal to the component. We _could_ place the spans on the outside, but it would result in empty spans whenever the icon wasn't rendered. Thoughts?

Reply:
Further to Slack chat - this looks good with the classes applied on all instances.